exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -1020,14 +1816,94 @@ func (e *Exchange) GetCurrencyTradeURL(_ context.Context, a asset.Item, cp curre
 	if err != nil {
 		return "", err
 	}
+	pairFormat, err := e.GetPairFormat(a, true)
+	if err != nil {
+		return "", err
+	}
 	switch a {
 	case asset.Spot:
-		cp.Delimiter = currency.UnderscoreDelimiter
-		return poloniexAPIURL + tradeSpot + cp.Upper().String(), nil
+		return mainURL + tradeSpotPath + pairFormat.Format(cp), nil
 	case asset.Futures:
-		cp.Delimiter = ""
-		return poloniexAPIURL + tradeFutures + cp.Upper().String(), nil
+		return mainURL + tradeFuturesPath + pairFormat.Format(cp), nil
 	default:
-		return "", fmt.Errorf("%w %q", asset.ErrNotSupported, a)
+		return "", fmt.Errorf("%w: %q", asset.ErrNotSupported, a)
+	}
+}
+
+// WebsocketSubmitOrder submits an order to the exchange via a websocket connection
+func (e *Exchange) WebsocketSubmitOrder(ctx context.Context, s *order.Submit) (*order.SubmitResponse, error) {
+	if err := s.Validate(e.GetTradingRequirements()); err != nil {
+		return nil, err
 	}
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	if err != nil {
+		return nil, err
+	}
+	if s.AssetType != asset.Spot {
+		return nil, fmt.Errorf("%w: %q", asset.ErrNotSupported, s.AssetType)
+	}
+	switch s.Type {
+	case order.Market, order.Limit, order.LimitMaker:
+	default:
+		return nil, fmt.Errorf("%w: %q", order.ErrUnsupportedOrderType, s.Type)
+	}
+	var stpMode string
+	switch s.TimeInForce {
+	case order.PostOnly:
+		stpMode = "EXPIRE_MAKER"
+	case order.GoodTillCancel:
+		stpMode = "EXPIRE_TAKER"
+	}
+	response, err := e.WsCreateOrder(ctx, &PlaceOrderRequest{
+		Symbol:                  s.Pair,
+		Price:                   s.Price,
+		Quantity:                s.Amount,
+		Amount:                  s.QuoteAmount,
+		AllowBorrow:             false,
+		Type:                    OrderType(s.Type),
+		Side:                    s.Side,
+		TimeInForce:             TimeInForce(s.TimeInForce),
+		ClientOrderID:           s.ClientOrderID,
+		SelfTradePreventionMode: stpMode,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return s.DeriveSubmitResponse(response.ID)
+}
+
+// WebsocketCancelOrder cancels an order via the websocket connection
+func (e *Exchange) WebsocketCancelOrder(ctx context.Context, req *order.Cancel) error {
+	if req.OrderID == "" && req.ClientOrderID == "" {
+		return order.ErrOrderIDNotSet
+	}
+	if err := req.Validate(req.StandardCancel()); err != nil {
+		return err
+	}
+	resp, err := e.WsCancelMultipleOrdersByIDs(ctx, []string{req.OrderID}, []string{req.ClientOrderID})
```

Comment:
Has no test coverage
If it does, please stop using `require.NoError` and use `assert.NoError` for multi-case tests